docs/userguide/008_The_Library_API.adoc

```diff
@@ -197,6 +196,69 @@ cycles.maxNumberToDetect=50
 cycles.maxNumberOfDependenciesPerEdge=5
 ----
 
+
+=== Modularization Rules
+
+[NOTE]
+Note: ArchUnit doesn't strive to be a "competition" for module systems like the
+Java Platform Module System. Such systems have advantages like checks at compile time
+versus test time as ArchUnit does. So, if another module system works well in your
+environment, there is no need to switch over. But ArchUnit can bring JPMS-like features
+to older code bases, etc. Java 8 projects, or environments where the JPMS is for some
+reason no option. It also can accompany a module system by adding additional rules e.g.
+on the API of a module.
+
+To express the concept of modularization ArchUnit offers `ArchModule`﻿s. The entrypoint into
+the API is `ModuleRuleDefinition`, e.g.
+
+[source,java,options="nowrap"]
+----
+ModuleRuleDefinition.modules().definedByPackages("..example.(*)..").shouldBeFreeOfCycles();
+----
+
+As the example shows, it shares some concepts with the <<Slices>> API. For example `definedByPackages(..)`
+follows the same semantics as `slices().matching(..)`. But, it also offers several powerful concepts
+beyond that API to express many different modularization scenarios.
+
+One example would be to express modules via annotation. We can introduce a custom annotation
+like `@AppModule` and follow a convention to annotate the top-level `package-info` file
+of each package we consider the root of a module. E.g.
+
+[source,java,options="nowrap"]
+.com/myapp/example/module_one/package-info.java
+----
+@AppModule(
+  name = "Module One",
+  allowedDependencies = {"Module Two", "Module Three"},
+  exposedPackages = {"..module_one.api.."}
+)
+package com.myapp.example.module_one;
+----
+
+We can then derive a general "self-growing" rule using the following API:
```

Comment:
What means _"self-growing"_?

Reply:
Mhmm, what I meant was that the rule automatically works for new components without the need to be adjusted. Take the PlantUML rule as a contrary example, every time you add a new module you have to adjust the diagram, i.e. the rule :man_shrugging: 
What I mean is that this can be set up once and teams can follow the convention without having to adjust the arch test itself...

Reply:
I just omitted the part, because I think it's evident enough from the following explanation and doesn't really contribute much to the docs :+1: